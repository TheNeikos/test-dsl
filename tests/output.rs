#[test]
fn check_invalid() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
            }
            tetcase {
            }
            foobar {
            }
            asd
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_unknown() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat 2 {
                    not_found
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_missing_argument() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_argument_wrong_type() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat hello {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}
