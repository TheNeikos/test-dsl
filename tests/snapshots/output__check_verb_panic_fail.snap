---
source: tests/output.rs
expression: "format!(\"{:?}\", miette::Error::new(tc.unwrap_err()))"
---
  [31mÃ—[0m Testcase did not run successfully
[31m  â”œâ”€â–¶ [0m  [31mÃ—[0m A panic occurred
[31m  â”‚   [0m   â•­â”€[3:17]
[31m  â”‚   [0m [2m2[0m â”‚                 testcase {
[31m  â”‚   [0m [2m3[0m â”‚ [35;1mâ•­[0m[35;1mâ”€[0m[35;1mâ–¶[0m                 foobar 2 {
[31m  â”‚   [0m [2m4[0m â”‚ [35;1mâ”‚[0m                       ofoo
[31m  â”‚   [0m [2m5[0m â”‚ [35;1mâ”œ[0m[35;1mâ”€[0m[35;1mâ–¶[0m                 }
[31m  â”‚   [0m   Â· [35;1mâ•°[0m[35;1mâ”€â”€â”€[0m[35;1mâ”€[0m [35;1min this node[0m
[31m  â”‚   [0m [2m6[0m â”‚                 }
[31m  â”‚   [0m   â•°â”€â”€â”€â”€
[31m  â”‚   [0m
[31m  â•°â”€â–¶ [0m  [31mÃ—[0m explicit panic
[31m      [0m
